lib/pbench/test/unit/server/query_apis/commons_get.py

```diff
@@ -0,0 +1,390 @@
+            date_range.append(f"{m.year:04}-{m.month:02}")
+        return date_range
+
+    def get_expected_status(self, payload: JSON, header: HeaderTypes) -> int:
+        """
+        Decode the various test cases we use, which are a combination of the
+        test case parametrization (for "user" parameter value) and the
+        parametrization of the `build_auth_header` fixture (for the API
+        authentication header);
+
+        1) If the "user" parameter is given and authentication is missing or
+           invalid, we expect UNAUTHORIZED (401) status;
+        2) If the "user" parameter is given and authenticated as a different
+           non-admin user with "access": "private", we expect FORBIDDEN (403)
+        3) If authenticated and a bad "user" parameter is given, we expect
+           NOT_FOUND (404)
+
+        Otherwise, we expect success
+        """
+        has_user = "user" in payload
+        u = payload.get("user")
+        a = payload.get("access")
+        unauthorized = not HeaderTypes.is_valid(header)
+        is_admin = header == HeaderTypes.VALID_ADMIN
+        if (has_user or a == Dataset.PRIVATE_ACCESS) and unauthorized:
+            expected_status = HTTPStatus.UNAUTHORIZED
+        elif u == "badwolf" and not unauthorized:
+            expected_status = HTTPStatus.NOT_FOUND
+        elif u != "drb" and a == Dataset.PRIVATE_ACCESS and not is_admin:
+            expected_status = HTTPStatus.FORBIDDEN
+        else:
+            expected_status = HTTPStatus.OK
+        return expected_status
+
+    @pytest.mark.parametrize(
+        "malformed_token",
+        ("malformed", "bear token" "Bearer malformed"),
+    )
+    def test_malformed_authorization_header(
+        self, client, server_config, malformed_token, attach_dataset
+    ):
+        """
+        Test behavior when the Authorization header is present but is not a
+        proper Bearer schema.
+
+        TODO: This actually tests behavior with no client authentication, as
+        Flask-HTTPTokenAuth hides the validation failure because our query
+        APIs select "optional" authentication. If we fix the authentication
+        validation to give a 401 (not 403) on bad/missing authentication token,
+        we should make the call with user/access parameters that would
+        otherwise be allowed for an unauthenticated client (e.g., all public
+        data) to distinguish the intent of this case from the path we're
+        actually hitting now, which appears to fail as expected but actually
+        fails because we're asking for {"access": "private"} data, which is not
+        allowed for an unauthenticated client call.
+        """
+        response = client.get(
+            server_config.rest_uri + self.pbench_endpoint,
+            headers={"Authorization": malformed_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.UNAUTHORIZED
```

Comment:
Can I? Maybe. Will I? More complicated... and I'd rather not.

This history is messy. There are several "authorization validation" tests originally written that turned out to have been "mostly meaningless" due to the way the Flask authorization package works. (E.g., bad authorization format is the same as none at all, as documented here.) I added the comments partly thinking we might want to write our own replacement for `Flask-HTTPAuth` to "do it right"... if we actually care... and as the time between then and now increases I'm less sure that we'll ever have any reason to care.

In any case, again, this is the same in `commons.py` and the new `commons_get.py` and I'd prefer not to have them diverge unnecessarily under this PR. (That is, I'd like the ability to `diff` them and see the `post` -> `get` transformation, which may or may not help with unifying them or layering a better model to avoid the duplication in the future.)

Reply:
OK.  On that basis, should I be insta-resolving the rest of my comments on `commons_get.py`?  (I'm kind of at a loss to tell what should be fixed in this PR and what should be deferred for the ease of merging.)

Reply:
Ha. The `POST` -> `GET` changes were required by Siddardh's change. We still have `ElasticBase` classes that use `POST` (and perhaps always will), and I directed Siddardh to start with a "brute force" clone that made just the changed he needed. In theory that makes it possible to compare them and evolve in parallel if necessary. I've always wanted to refactor `Commons` just because a lot has changed (and been learned, such as the broken validation tests) since it was written. Realistically, this might sit on the back burner until we build a real unit test infrastructure for the API classes instead of using the mocked requests/Flask... but who knows?

Mostly, I was pointing out that most if not all of your comments here could be against `commons.py` as well and if we're taking action on them we should do it in both at the same time (if they can't ultimately be unified) and that suggests at least deferring to another PR. 🍤

Reply:
Alright, I resolved a bunch of them, mostly the semi-substantial ones.  The trivial ones should probably be fixed in this PR, and then, when someone diffs the two modules, they will show up as small things to fix in the other one.  I did leave two of them unresolved which call for more substantive changes -- there's a redundant assertion (which should be trivial to get rid of and easy to merge) and there's the save-and-restore thing which I think should be changed to a copy-and-destroy approach.